src/RoomsSDKAdapter.js

```diff
@@ -134,4 +144,119 @@ export default class RoomsSDKAdapter extends RoomsAdapter {
 
     return this.getRoomObservables[ID];
   }
+
+  /**
+   * Invoke request to listActivities on adapter
+   *
+   * @param {object} Params to send to conversation api method
+   * @returns
+   * @private
+   */
+  fetchActivities({roomId, earliestActivityDate}) {
+    const {limit} = this;
+    const conversationId = deconstructHydraId(roomId).id;
+
+    logger.debug('ROOM', 'fetchActivities()', ['called with', {conversationId, limit, earliestActivityDate}]);
+
+    let maxDate = earliestActivityDate;
+
+    if (maxDate === null) {
+      maxDate = undefined;
+    }
+
+    return this.datasource.internal.conversation.listActivities({
+      conversationId,
+      limit,
+      lastActivityFirst: true,
+      maxDate,
+    });
+  }
+
+  /**
+   * Handles checking if there is more activities to fetch.
+   *
+   * @param {string} roomId The room id to check
+   * @returns {boolean} True if there is more activities
+   */
+  hasMoreActivities(roomId) {
+    const {earliestActivityDate, activityIds} = this.roomActivities.get(roomId);
+
+    if (activityIds[earliestActivityDate]) {
+      return activityIds[earliestActivityDate] === this.limit;
+    }
+
+    return true;
+  }
+
+  /**
+   * Fetches past activities and returns array of (id, published) objects.
+   *
+   * @param {string} roomId The id of the room
+   * @param {number} limit The max number of activities to return
+   * @returns {external:Observable.<Activity>} Observable stream that emits activity data
+   */
+  async fetchPastActivities(roomId) {
```

Comment:
`fetchPastActivities` is not [part of the interface](https://github.com/webex/component-adapter-interfaces/blob/master/src/RoomsAdapter.js#L67) and therefore cannot be public. Where is this method called by the public methods?

Reply:
Spicy